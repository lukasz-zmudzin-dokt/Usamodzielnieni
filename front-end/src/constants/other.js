const DEFAULT_INPUT = "-- Wybierz --";

export { DEFAULT_INPUT };
