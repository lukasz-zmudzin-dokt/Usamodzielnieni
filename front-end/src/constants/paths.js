const paths = {
    DASHBOARD: "/",
    CV_CREATOR: "/cvEditor",
    CV_EDITOR: "/cvEditor/:id",
    REGISTER: "/newAccount/",
    REGISTER_ADMIN: "/newAccount/:role(staff)",
    FOOTER: "/footer",
    LOGIN: "/login",
    USER: "/user",
    OFFER_FORM: "/offerForm/:id?",
    JOB_OFFERS: "/jobOffers",
    CV_APPROVAL: "/cvApproval",
    MY_OFFERS: "/myOffers",
    CONTACT_PAGE: "/contact",
    BLOG_POST: "/blog/blogpost/:id",
    BLOG_PAGE: "/blog",
    BLOG_FORM: "/blog/newPost",
    BLOG_EDIT: "/blog/newPost/:id",
    JOB_OFFER_DETAILS: "/jobOffers/:id",
    MY_CVS: "/myCVs",
    USER_APPROVAL: "/userApproval",
    OFFER_APPROVAL: "/offerApproval"
};
export {paths};