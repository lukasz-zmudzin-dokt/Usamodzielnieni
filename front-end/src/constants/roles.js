import { staffTypes } from "./staffTypes";

export const adminGroup = [
  {
    name: staffTypes.VERIFICATION,
    placeholder: "Weryfikacja użytkowników",
  },
  {
    name: staffTypes.CV,
    placeholder: "Weryfikacja CV",
  },
  {
    name: staffTypes.JOBS,
    placeholder: "Weryfikacja ofert pracy",
  },
  {
    name: staffTypes.BLOG_CREATOR,
    placeholder: "Kreator postów na blogu",
  },
  {
    name: staffTypes.BLOG_MODERATOR,
    placeholder: "Moderator bloga / designer",
  },
  {
    name: staffTypes.GUEST,
    placeholder: "Konto gościa",
  },
  {
    name: staffTypes.CHAT,
    placeholder: "Dostęp do czatu",
  },
];

export const commonGroup = [
  {
    name: "Podopiecznym",
  },
  {
    name: "Pracodawcą",
  },
];
