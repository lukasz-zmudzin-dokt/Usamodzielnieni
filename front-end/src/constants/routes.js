import {
  UserProfilePage,
  LoginPage,
  RegisterPage,
  Footer,
  CVEditorPage,
  Menu,
  OfferForm,
  MyOffersPage,
  CVApprovalPage
} from "Views";

const paths = {
  DASHBOARD: "/",
  CV_EDITOR: "/cvEditor",
  REGISTER: "/newAccount",
  FOOTER: "/footer",
  LOGIN: "/login",
  USER: "/user",
  OFFER_FORM: "/offerForm",
  JOB_OFFERS: "/jobOffers",
  CV_APPROVAL: "/cvApproval",
  MY_OFFERS: "/myOffers"
};

export const userTypes = {
  STANDARD: "Standard",
  STAFF: "Staff",
  EMPLOYER: "Employer"
};

export default [
  {
    path: paths.DASHBOARD,
    component: Menu,
    exact: true
  },
  {
    path: paths.CV_EDITOR,
    component: CVEditorPage,
    isPrivate: true,
    type: userTypes.STANDARD
  },
  {
    path: paths.REGISTER,
    component: RegisterPage,
    exact: true
  },
  {
    path: paths.FOOTER,
    component: Footer,
    exact: true
  },
  {
    path: paths.LOGIN,
    component: LoginPage,
    exact: true
  },
  {
    path: paths.USER,
    component: UserProfilePage,
    isPrivate: true,
    type: undefined // jeżeli jest undefined to znaczy że jest dostępne dla wszystkich typów konta
  },
  {
    path: paths.JOB_OFFERS,
    component: UserProfilePage, // tu trzeba zmienić komponent
    isPrivate: true,
    type: userTypes.EMPLOYER
  },
  {
    path: paths.OFFER_FORM,
    component: OfferForm,
    isPrivate: true,
    type: userTypes.EMPLOYER
  },
  {
    path: paths.CV_APPROVAL,
    component: CVApprovalPage,
    isPrivate: true,
    type: userTypes.STAFF
  },
  {
    path: paths.MY_OFFERS,
    component: MyOffersPage,
    isPrivate: true,
    type: userTypes.EMPLOYER
  }
];

export { paths };
