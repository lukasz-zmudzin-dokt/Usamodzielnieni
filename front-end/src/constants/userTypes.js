export const userTypes = {
  STANDARD: "standard",
  STAFF: "staff",
  EMPLOYER: "employer",
};
