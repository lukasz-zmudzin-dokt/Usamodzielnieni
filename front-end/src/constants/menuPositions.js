import { userTypes } from "./userTypes";
import { staffTypes } from "./staffTypes";
import { paths } from "./paths";

export default [       
    {
        name: "Kreator CV",
        path: paths.CV_CREATOR,
        allowed: [userTypes.STANDARD],
        verified: true
    },
    {
        name: "Moje CV",
        path: paths.MY_CVS,
        allowed: [userTypes.STANDARD],
        verified: true
    },
    {
        name: "Akceptacja CV",
        path: paths.CV_APPROVAL,
        allowed: [staffTypes.CV],
        verified: true
    },
    {
        name: "Oferty pracy",
        path: paths.JOB_OFFERS,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER]
    },
    {
        name: "Dodaj ofertę",
        path: paths.OFFER_FORM,
        allowed: [userTypes.EMPLOYER],
        verified: true
    },
    {
        name: "Moje oferty",
        path: paths.MY_OFFERS,
        allowed: [userTypes.EMPLOYER],
        verified: true
    },
    {
        name: "Blogi",
        path: paths.BLOG_PAGE,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER]
    },
    {
        name: "Dodaj post",
        path: paths.BLOG_FORM,
        allowed: [staffTypes.BLOG_CREATOR],
        verified: true
    },
    {
        name: "Telefony",
        path: paths.CONTACT_PAGE,
        allowed: undefined, //wszyscy są upoważnieni - zalogowani i niezalogowani
    },
    {
        name: "Kroki",
        path: paths.STEPS,
        allowed: undefined //wszyscy są upoważnieni - zalogowani i niezalogowani
    },
    {
        name: "Chat",
        path: paths.CHATS,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER],
        verified: true,
    }
];
