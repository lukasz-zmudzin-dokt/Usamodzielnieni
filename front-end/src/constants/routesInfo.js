import { userTypes } from "./userTypes";
import { staffTypes } from "./staffTypes";
import { paths } from "./paths";

export default [
  {
    path: paths.DASHBOARD,
    exact: true,
  },
  {
    path: paths.CV_CREATOR,
    isPrivate: true,
    type: [userTypes.STANDARD, userTypes.STAFF],
    group: [staffTypes.GUEST, staffTypes.BLOG_MODERATOR],
    exact: true,
    userVerified: true,
  },
  {
    path: paths.CV_EDITOR,
    isPrivate: true,
    type: userTypes.STANDARD,
    exact: true,
    userVerified: true,
  },
  {
    path: paths.REGISTER,
    exact: true,
  },
  {
    path: paths.REGISTER_ADMIN,
    exact: true,
    isPrivate: true,
    type: userTypes.STAFF,
    group: staffTypes.VERIFICATION,
    userVerified: true,
  },
  {
    path: paths.FOOTER,
    exact: true,
  },
  {
    path: paths.LOGIN,
    exact: true,
  },
  {
    path: paths.USER,
    isPrivate: true,
    type: undefined, // jeżeli jest undefined to znaczy że jest dostępne dla wszystkich typów konta
  },
  {
    path: paths.JOB_OFFERS,
    exact: true,
    type: undefined,
  },
  {
    path: paths.JOB_OFFER_DETAILS,
    exact: true,
    type: undefined,
  },
  {
    path: paths.OFFER_FORM,
    isPrivate: true,
    type: [userTypes.EMPLOYER, userTypes.STAFF],
    group: [staffTypes.GUEST],
    userVerified: true,
  },
  {
    path: paths.CV_APPROVAL,
    isPrivate: true,
    type: userTypes.STAFF,
    group: staffTypes.CV,
    userVerified: true,
  },
  {
    path: paths.MY_OFFERS,
    isPrivate: true,
    type: userTypes.EMPLOYER,
    userVerified: true,
  },
  {
    path: paths.CONTACT_PAGE,
  },
  {
    path: paths.BLOG_POST,
    exact: true,
  },
  {
    path: paths.BLOG_PAGE,
    exact: true,
  },
  {
    path: paths.BLOG_FORM,
    isPrivate: true,
    type: userTypes.STAFF,
    group: [staffTypes.BLOG_CREATOR],
    exact: true,
    userVerified: true,
  },
  {
    path: paths.BLOG_EDIT,
    isPrivate: true,
    type: userTypes.STAFF,
    group: staffTypes.BLOG_CREATOR,
    exact: true,
    userVerified: true,
  },
  {
    path: paths.MY_CVS,
    isPrivate: true,
    type: userTypes.STANDARD,
    userVerified: true,
  },
  {
    path: paths.USER_APPROVAL,
    isPrivate: true,
    type: userTypes.STAFF,
    group: staffTypes.VERIFICATION,
    userVerified: true,
  },
  {
    path: paths.STEPS,
    isPrivate: true,
  },
  {
    path: paths.CV_CORRECTION,
    isPrivate: true,
    type: userTypes.STAFF,
    group: staffTypes.CV,
  },
  {
    path: paths.CHATS,
    isPrivate: true,
    type: undefined,
    group: staffTypes.CHAT,
    exact: true,
    userVerified: true,
  },
  {
    path: paths.CHAT_DETAILS,
    isPrivate: true,
    type: undefined,
    group: staffTypes.CHAT,
    exact: true,
    userVerified: true,
  },
  {
    path: paths.USER_LIST,
    isPrivate: true,
    type: userTypes.STAFF,
    exact: true,
    userVerified: true,
  },
  {
    path: paths.OFFER_APPROVAL,
    isPrivate: true,
    type: userTypes.STAFF,
    group: staffTypes.JOBS,
  },
  {
    path: paths.CHANGE_DATA,
    isPrivate: true,
    type: userTypes.STAFF,
    group: undefined,
  },
  {
    path: paths.PASSWORD_RESET,
    type: undefined,
    exact: true,
  },
  {
    path: paths.NEW_PASSWORD,
    type: undefined,
    exact: true,
  },
  {
    path: paths.BLOG_PAGE_FILTERED,
    type: undefined,
    exact: true
  }
];
