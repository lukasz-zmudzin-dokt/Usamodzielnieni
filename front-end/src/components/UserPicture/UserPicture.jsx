import React from 'react';
import { userTypes } from 'constants/userTypes';

const UserPicture = ({ user, ...rest }) => {
    const getColorClassSuffix  = () => {
        if (user.type === userTypes.STAFF && user.data?.group_type?.length) {
            if (user.data.group_type.length > 1) {
                return 'multi';
            }
            return user.data.group_type[0];
        }
        return 'standard';
    }
    
    const getFirstLetters = () => {
        if (user.data?.first_name?.length && user.data?.last_name?.length) {
            return user.data.first_name[0] + user.data.last_name[0];
        }
        return '?';
    }

    return (
        <div className="userPicture__container" {...rest}>
            <div className={`userPicture userPicture--${getColorClassSuffix()}`}>
                <span>{ getFirstLetters() }</span>
            </div>
        </div>
    )
}

export default UserPicture;
