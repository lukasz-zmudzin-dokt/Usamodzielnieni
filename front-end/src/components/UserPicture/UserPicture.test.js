import React from "react";
import { render } from "@testing-library/react";
import UserPicture from "./UserPicture";
import { userTypes } from "constants/userTypes";

describe("UserPicture", () => {
  let user;

  beforeEach(() => {
    user = {
      type: userTypes.STANDARD,
      data: {
        first_name: "Jan",
        last_name: "Kowalski",
      },
    };
  });

  it("should render without crashing", () => {
    const { container } = render(<UserPicture user={user} />);

    expect(container).toMatchSnapshot();
  });

  it("should render staff picture", () => {
    user.type = "Staff";
    user.data.group_type = ["jakis_group_type"];

    const { container } = render(<UserPicture user={user} />);

    expect(container).toMatchSnapshot();
  });

  it("should render multi staff picture", () => {
    user.type = userTypes.STAFF;
    user.data.group_type = ["jakis_group_type", "jakiÅ›_inny_type"];

    const { container } = render(<UserPicture user={user} />);

    expect(container).toMatchSnapshot();
  });

  it("should render without crashing when data is undefined", () => {
    user.data = undefined;

    const { container } = render(<UserPicture user={user} />);

    expect(container).toMatchSnapshot();
  });
});
