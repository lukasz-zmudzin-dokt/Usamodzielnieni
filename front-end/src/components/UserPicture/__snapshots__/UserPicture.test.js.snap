// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPicture should render multi staff picture 1`] = `
<div>
  <div
    class="userPicture__container"
  >
    <div
      class="userPicture userPicture--multi"
    >
      <span>
        JK
      </span>
    </div>
  </div>
</div>
`;

exports[`UserPicture should render staff picture 1`] = `
<div>
  <div
    class="userPicture__container"
  >
    <div
      class="userPicture userPicture--jakis_group_type"
    >
      <span>
        JK
      </span>
    </div>
  </div>
</div>
`;

exports[`UserPicture should render without crashing 1`] = `
<div>
  <div
    class="userPicture__container"
  >
    <div
      class="userPicture userPicture--standard"
    >
      <span>
        JK
      </span>
    </div>
  </div>
</div>
`;

exports[`UserPicture should render without crashing when data is undefined 1`] = `
<div>
  <div
    class="userPicture__container"
  >
    <div
      class="userPicture userPicture--standard"
    >
      <span>
        ?
      </span>
    </div>
  </div>
</div>
`;
